src/Util/GeometryUtil/GeometryUtil.js

```diff
@@ -163,6 +163,32 @@ class GeometryUtil {
       // Logger.warn('Can not merge mixed geometries into one multigeometry.');
       return undefined;
     }
+
+    // split all multi-geometries to simple ones if passed geometries are
+    // multigeometries
+    const multiGeomPrefix = 'Multi';
+    if (geomType.startsWith(multiGeomPrefix)) {
+      const multiGeomPartType = geomType.substring(multiGeomPrefix.length);
+      let subGeometries = [];
+      geometries.forEach(geometry => {
+        switch (multiGeomPartType) {
+          case 'Polygon':
+            subGeometries.push(...geometry.getPolygons());
+            break;
+          case 'LineString':
+            subGeometries.push(...geometry.getLineStrings());
+            break;
+          case 'Point':
+            subGeometries.push(...geometry.getPoints());
+            break;
+          default:
+            break;
+        }
+      });
+      geometries = subGeometries;
+      geomType = multiGeomPartType;
+    }
```

Comment:
I suggest to move this code to an own method as this might be needed in other util methods inside the `GeometryUtil` as well.

Reply:
Alright, I'll create an issue for this.